### PR TITLE
Printer cheatcode

### PR DESCRIPTION
**Pull request by DarkaMaul**
# Change description

Implement a new printer to display cheatcodes usage.
Fixes https://github.com/crytic/slither/issues/2141

# Caveats

The cheatcodes are usually used in test files. Thus, the classic command:

```shell
slither --printer cheatcode project_directory/
```

will not work because the default is to remove `tests` files from being analyzed.

To force the usage:
```shell
slither --printer cheatcode --foundry-compile-all project_directory
```

# Example output
```shell
slither --print cheatcode --foundry-compile-all tests/e2e/printers/test_data/test_printer_cheatcode
```

```console
'forge clean' running (wd: ./slither/tests/e2e/printers/test_data/test_printer_cheatcode)
'forge build --build-info' running (wd: ./slither/tests/e2e/printers/test_data/test_printer_cheatcode)
INFO:Printers:CounterTest (test/Counter.t.sol)
	setUp
		L[21]: deal
		L[22]: deal
	testIncrement
		L[28]: prank
		L[30]: assertEq
		L[32]: prank
		L[34]: assertEq

INFO:Slither:tests/e2e/printers/test_data/test_printer_cheatcode analyzed (34 contracts)
```

# Missing

- [ ] Wiki entry
- [ ] Force to analyze test file if the printer is selected?

## Summary by CodeRabbit

- **New Features**
  - Introduced a Solidity smart contract named `Counter` with functions for number manipulation.
  - Added a `CounterTest` suite to test the functionality of the `Counter` contract.

- **Documentation**
  - Included a `README.md` file with instructions for initializing a counter using a specific command.
  - Added a `foundry.toml` file with configuration settings for the `test_printer_cheatcode` printer.

- **Tests**
  - Updated test script with new imports, global variables, and a test function for `CheatcodePrinter`.

### CODE REVIEW

File: slither/printers/summary/cheatcodes.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            is_test_var = contract.variables_as_dict.get("IS_TEST", None)
+            is_test = False
+            if is_test_var is not None:
+                try:
+                    is_test = is_test_var.expression.value == "true"
+                except AttributeError:
+                    pass
+
+            if not is_test:
+                continue
```
Comment (DarkaMaul) [inline slither/printers/summary/cheatcodes.py:42]: Do we also want to flag cheatcode issue in non test files?

Comment (0xalpharush) [reply slither/printers/summary/cheatcodes.py:42]: I don't think so

File: slither/printers/summary/cheatcodes.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                    for op in node.all_slithir_operations():
+                        if (
+                            isinstance(op, HighLevelCall)
+                            and op.function.contract == vm
+                            and op.function.visibility == "external"
+                        ):
+                            found_function = True
+                            function_info += (
+                                f"\t\tL{node.source_mapping.lines}: {op.function.name}\n"
+                            )
```
Comment (0xalpharush) [inline slither/printers/summary/cheatcodes.py:59]: Can you use `to_detailed_str()` here please? And maybe show the source mapping content?

Comment (DarkaMaul) [reply slither/printers/summary/cheatcodes.py]: New output is like this

```
INFO:Printers:CounterTest (test/Counter.t.sol)
	setUp
		deal - (test/Counter.t.sol#21 (9 - 32)
		vm.deal(alice,1000000000000000000)

		deal - (test/Counter.t.sol#22 (9 - 30)
		vm.deal(bob,2000000000000000000)

	testIncrement
		prank - (test/Counter.t.sol#28 (9 - 24)
		vm.prank(alice)

		assertEq - (test/Counter.t.sol#30 (9 - 38)
		assertEq(counter.number(),1)

		prank - (test/Counter.t.sol#32 (9 - 22)
		vm.prank(bob)

		assertEq - (test/Counter.t.sol#34 (9 - 38)
		assertEq(counter.number(),2)
```

I'm open to any suggestion to enhance it.

**Pull request merged**